OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -452,7 +452,15 @@ public void onSessionEnding(@NonNull List<OSInfluence> lastInfluences) {
    @Nullable private static OSOutcomeEventsController outcomeEventsController;
    @Nullable private static OSOutcomeEventsFactory outcomeEventsFactory;
 
-   private static AdvertisingIdentifierProvider mainAdIdProvider = new AdvertisingIdProviderGPS();
+   @Nullable private static AdvertisingIdentifierProvider adIdProvider;
+   private static synchronized @Nullable AdvertisingIdentifierProvider getAdIdProvider() {
+      if (adIdProvider == null) {
+         if (OSUtils.isAndroidDeviceType())
```

Comment:
maybe put on the same if?

Reply:
I set it up nested like this as we may want to add an `else if (OSUtils.isHuaweiDeviceType()` ect if we can support other ad providers in the future.

Reply:
sounds good!